File: src/main/java/com/fithub/model/Payment.java
package com.fithub.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "payments")
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne
    private Member member;
    
    private BigDecimal amount;
    private LocalDate dueDate;
    private LocalDate paymentDate;
    
    @Enumerated(EnumType.STRING)
    private PaymentStatus status;
    
    @Enumerated(EnumType.STRING)
    private PaymentType type;
    
    private String description;
}

File: types/payment.ts
export type PaymentStatus = "pending" | "paid" | "overdue";

export interface Member {
  id: string;
  name: string;
  email: string;
  phone: string;
}

export interface Payment {
  id: string;
  memberId: string;
  member: Member;
  type: string;
  amount: number;
  dueDate: string;
  status: PaymentStatus;
  createdAt: string;
  updatedAt: string;
}